File: 0965-unique-email-addresses/0965-unique-email-addresses.cpp
class Solution {
public:
    int numUniqueEmails(vector<string>& emails) {
        std::unordered_set<string> seen;
        string local;
        string domain;

        for(auto& email: emails){
            int pos = email.find("@");
            local = email.substr(0, pos);
            domain = email.substr(pos+1);

            string find;
            bool ignore = false;
            for(auto& ch : local){

                if(ch == '+'){
                ignore = true;
                break;
                }else if(!ignore && ch != '.'){
                    find += ch;
                }
            }

            string found = find + "@" + domain;
            seen.insert(found);
            
            
        }





        return seen.size();

        
    }
};

File: 0965-unique-email-addresses/0965-unique-email-addresses.py
class Solution:
    def numUniqueEmails(self, emails: List[str]) -> int:
        seen = set()


        for email in emails:

            res = ""
            domain  = ""
            seenPlus = False
            seenAt = False

            for j in range(len(email)):

                if email[j]  == "@": seenAt = True

                if email[j] == "+": seenPlus = True

                if not seenPlus and not seenAt and email[j] != "." and email[j] != "+":
                    res += email[j]
                


                

                if seenAt:
                    domain = email[j +1 :]
                    break
            

            print(domain, res)
            seen.add((res, domain))

        return len(seen)

        
        